src/ActivationFunctionInterface.hpp

```diff
@@ -25,6 +25,18 @@ class ActivationFunctionInterface
 
     // third derivative of the activation function
     virtual double f3d(const double &) = 0;
+
+    // function to calculate function value and all needed derivatives together, allowing for speedup over individual calls
+    virtual void fad(const double &in, double &v, double &v1d, double &v2d, double &v3d, const bool flag_d1 = false, const bool flag_d2 = false, const bool flag_d3 = false)
```

Comment:
I am not sure here, so it is mainly a question.
But why should this function be virtual?